perftest/simulations/src/gatling/scala/org/projectnessie/perftest/gatling/KeyListSpillingSimulation.scala

```diff
@@ -0,0 +1,244 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.perftest.gatling
+
+import io.gatling.core.Predef._
+import io.gatling.core.scenario.Simulation
+import io.gatling.core.structure.{ChainBuilder, ScenarioBuilder}
+import org.projectnessie.error.NessieReferenceNotFoundException
+import org.projectnessie.model.Operation.Put
+import org.projectnessie.model._
+import org.projectnessie.perftest.gatling.Predef.nessie
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.concurrent.duration.{FiniteDuration, SECONDS}
+import scala.jdk.CollectionConverters._
+
+/**
+ * Measure throughput of incremental put commits against an existing HEAD with a preexisting live keylist.
+ *
+ * @see [[KeyListSpillingParams]]
+ */
+class KeyListSpillingSimulation extends Simulation {
+
+  private val log = LoggerFactory.getLogger(classOf[KeyListSpillingSimulation])
+
+  /**
+   * Fixed-size key padding (256 chars, repeating every 8 chars)
+   */
+  private val paddingBuilder: mutable.StringBuilder = new mutable.StringBuilder()
+  for (i <- 1 to 32) {
+    paddingBuilder.append("abcdefg_")
+  }
+  val padding: String = paddingBuilder.toString();
+
+  /**
+   * Configuration specific to this simulation and scenario.
+   */
+  private val params: KeyListSpillingParams = KeyListSpillingParams.fromSystemProperties()
+
+  /**
+   * Session key for the zero-indexed commit number associated with the current simulated Gatling user.
+   *
+   * Gatling initializes and increments this for us via [[repeat]].
+   */
+  private val commitIndexKey = "commitIndex"
+
+  /**
+   * Commit once to a branch conveyed via the [[Session]].
+   *
+   * Although the returned chain fragment will commit only once, the number of put operations in that commit is
+   * determined by [[KeyListSpillingParams.putsPerTestCommit]].
+   */
+  private def commitToBranch: ChainBuilder = {
+      exec(
+        nessie("TestBranchCommit")
+          .execute { (client, session) =>
+
+            // This is initialized and incremented on our behalf by Gatling (see buildScenario)
+            val commitIndex = session(commitIndexKey).as[Int]
+            val commitNum = commitIndex + 1
+            val testBranch = session("branch").as[Branch]
+            val userId = session.userId
+            val putOps: List[Operation] = List.range(0, params.putsPerTestCommit).map(
+              i => {
+                val key = ContentKey.of(s"${userId} ${commitIndex} ${i}" + padding)
```

Comment:
Nit: the enclosing `{}` aren't necessary here (and in other strings belos)

Reply:
I think I've gotten all instances of this, but left it in where underscores joined separate variable names.  I'm not sure about how the interpolator resolves those, but I thought it might be more legible regardless.  I'd be happy to change it though.